analysis/parameter_tuning.py

```diff
@@ -111,27 +116,32 @@ class TuneResult:
     utility_reports: List[metrics.UtilityReport]
 
 
+class ParametersSearchStrategy(Enum):
+    QUANTILES = 1
+    CONSTANT_RELATIVE_STEP = 2
+
+
 def _find_candidate_parameters(
         hist: histograms.DatasetHistograms,
         parameters_to_tune: ParametersToTune,
-        metric: pipeline_dp.Metrics) -> analysis.MultiParameterConfiguration:
+        metric: pipeline_dp.Metrics,
+        strategy: ParametersSearchStrategy = ParametersSearchStrategy.QUANTILES,
+        n_max: int = None) -> analysis.MultiParameterConfiguration:
```

Comment:
It's better to specify `max_candidates` from the caller code and that this function would detect how many to generate candidates by each coordinate. 
Example
`max_candidates = 25`
`max_candidates = 25`
  if`
  if` parameters_to_tune.max_partitions_contributed == parameters_to_tune.max_contributions_per_partition == True`
then it takes by 5 candidate by  l0 and linf
but if "max_partitions_contributed == True and max_contributions_per_partition == False` then it takes
25 options for l0.

WDYT?

Reply:
yes, nice idea, implemented and made small refactoring which I think makes code more readable.